rero_ils/modules/collections/api.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""API for manipulating collections."""
+
+from functools import partial
+
+from .models import CollectionIdentifier
+from ..api import IlsRecord, IlsRecordsIndexer, IlsRecordsSearch
+from ..fetchers import id_fetcher
+from ..minters import id_minter
+from ..providers import Provider
+
+# provider
+CollectionProvider = type(
+    'CollectionProvider',
+    (Provider,),
+    dict(identifier=CollectionIdentifier, pid_type='coll')
+)
+# minter
+collection_id_minter = partial(id_minter, provider=CollectionProvider)
+# fetcher
+collection_id_fetcher = partial(id_fetcher, provider=CollectionProvider)
+
+
+class CollectionsSearch(IlsRecordsSearch):
+    """CollectionsSearch."""
+
+    class Meta:
+        """Search only on collection index."""
+
+        index = 'collections'
+        doc_types = None
+
+
+class Collection(IlsRecord):
+    """Collection class."""
+
+    minter = collection_id_minter
+    fetcher = collection_id_fetcher
+    provider = CollectionProvider
+
```

Comment:
do you need to define `pids_exist_check` here.? see https://github.com/rero/rero-ils/blob/50486c81df2f42b0f15457434d78afe23c8dc0d0/rero_ils/modules/holdings/api.py#L85 as an example

Reply:
Done